src/Common/ProgressIndication.cpp

```diff
@@ -99,8 +99,8 @@ void ProgressIndication::writeFinalProgress()
     if (elapsed_ns)
         std::cout << " (" << formatReadableQuantity(progress.read_rows * 1000000000.0 / elapsed_ns) << " rows/s., "
                     << formatReadableSizeWithDecimalSuffix(progress.read_bytes * 1000000000.0 / elapsed_ns) << "/s.)";
-    else
-        std::cout << ". ";
+    auto peak_memory_usage = getMemoryUsage().peak;
```

Comment:
I have one concern here how a new client works with an old server. Client would print `0` as `peak_memory_usage`

Reply:
It seems to me that this entire line is for information, and it is unlikely that the client will rely on this value. 
Try not to display the value for old servers?

Reply:
It would right and heavy. May be just enough not to log it when it zero. Just because memory usage never is zero.

Reply:
Added check (not show if old server). I use '-1' (as default) because memory  can be zero (I don't know why it works like this; on small queries always 0).